addons/website_portal_sale/models/sale_order.py

```diff
@@ -8,6 +8,12 @@ class SaleOrder(models.Model):
 
     _inherit = 'sale.order'
 
+    website_url = fields.Char('Website URL', compute='_website_url', help='The full URL to access the document through the website.')
+
+    def _website_url(self):
```

Comment:
Not all orders are accessible. Shouldn't we find a way to compute the info only once, with get access action and website url ? (This remark apply to other objects by the way :) )